.happy/terraform/modules/sfn_config/nextstrain-ondemand.wdl

```diff
@@ -0,0 +1,114 @@
+        aws_region = aws_region,
+        aspen_config_secret_name = aspen_config_secret_name,
+        remote_dev_prefix = remote_dev_prefix,
+        workflow_id = workflow_id,
+        s3_filestem = s3_filestem,
+    }
+}
+
+
+task nextstrain_workflow {
+    input {
+        String docker_image_id
+        String aws_region
+        String aspen_config_secret_name
+        String remote_dev_prefix
+        Int    workflow_id
+        String s3_filestem
+    }
+
+    command <<<
+    set -Eeuxo pipefail
+    shopt -s inherit_errexit
+
+    df 1>&2
+    cat /proc/meminfo 1>&2
+
+    start_time=$(date +%s)
+    build_id=$(date +%Y%m%d-%H%M)
+
+    export ASPEN_CONFIG_SECRET_NAME=~{aspen_config_secret_name}
+    if [ "~{remote_dev_prefix}" != "" ]; then
+        export REMOTE_DEV_PREFIX="~{remote_dev_prefix}"
+    fi
+
+    aws configure set region ~{aws_region}
+
+    # fetch aspen config
+    aspen_config="$(aws secretsmanager get-secret-value --secret-id ~{aspen_config_secret_name} --query SecretString --output text)"
+    aspen_s3_db_bucket="$(jq -r .S3_db_bucket <<< "$aspen_config")"
+
+    # set up ncov
+    mkdir -p ncov/my_profiles/aspen ncov/results
+    (cd ncov &&
+     git init &&
+     git fetch --depth 1 git://github.com/nextstrain/ncov.git &&
+     git checkout FETCH_HEAD
+    )
+    ncov_git_rev=$(cd ncov && git rev-parse HEAD)
+
+    cp /usr/src/app/aspen/workflows/nextstrain_run/nextstrain_profile/* ncov/my_profiles/aspen/
+
+    # dump the sequences, metadata, and builds.yaml for a run out to disk.
+    aligned_gisaid_location=$(
+        python3 /usr/src/app/aspen/workflows/nextstrain_run/export.py \
+               --phylo-run-id "~{workflow_id}"                        \
+               --county-sequences ncov/data/sequences_aspen.fasta     \
+               --county-metadata ncov/data/metadata_aspen.tsv         \
+               --sequences ncov/data/sequences_selected.fasta         \
+               --metadata ncov/data/metadata_selected.tsv             \
+               --builds-file ncov/my_profiles/aspen/builds.yaml       \
+    )
+    aligned_gisaid_s3_bucket=$(echo "${aligned_gisaid_location}" | jq -r .bucket)
+    aligned_gisaid_sequences_s3_key=$(echo "${aligned_gisaid_location}" | jq -r .sequences_key)
+    aligned_gisaid_metadata_s3_key=$(echo "${aligned_gisaid_location}" | jq -r .metadata_key)
+
+    # fetch the gisaid dataset
+    aws s3 cp --no-progress "s3://${aligned_gisaid_s3_bucket}/${aligned_gisaid_sequences_s3_key}" - | zstdmt -d | xz -2 > ncov/results/aligned_gisaid.fasta.xz
+    aws s3 cp --no-progress "s3://${aligned_gisaid_s3_bucket}/${aligned_gisaid_metadata_s3_key}" ncov/data/metadata_gisaid.tsv
+
+    (cd ncov && snakemake --printshellcmds auspice/ncov_aspen.json --profile my_profiles/aspen/ --resources=mem_mb=312320) || aws s3 cp ncov/.snakemake/log/ "s3://${aspen_s3_db_bucket}/phylo_run/${build_id}/" --recursive
+
+    # upload the tree to S3
+    key="phylo_run/${build_id}/~{s3_filestem}.json"
```

Comment:
Can/Should we add a workflow ID to this path? Multiple jobs that start at the ~same time could clobber each other's result files.

Reply:
I updated the s3 paths.